app/src/test/java/com/eis/geoCalendar/database/AbstractEventDatabaseTest.java

```diff
@@ -0,0 +1,183 @@
+     */
+    @After
+    public void clearInstances() {
+        clearActiveInstances();
+    }
+
+    /**
+     * This method is necessary, due to an issue with MySQL, where static instances of the
+     * database cannot be reset between tests automatically.
+     */
+    public static void clearActiveInstances() {
+        try {
+            //Getting the field named "activeInstances" (the map with all active databases)
+            Field instance = AbstractEventDatabase.class.getDeclaredField("activeInstances");
+            //Making it accessible
+            instance.setAccessible(true);
+            //Casting it to a Map
+            Map<?, ?> privateField = (Map<?, ?>) instance.get(null);
+            //Making it empty
+            if (privateField != null)
+                privateField.clear();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Testing whether two instances with the same name are actually the same instance.
+     */
+    @Test
+    public void poolSameName() {
+        AbstractEventDatabase anotherDatabase =
+                AbstractEventDatabase.getInMemoryInstance(RuntimeEnvironment.systemContext,
+                        DEFAULT_DB_NAME);
+        assertEquals(database, anotherDatabase);
+    }
+
+    /**
+     * Testing whether two instances with a different name are different instances.
+     */
+    @Test
+    public void poolDifferentName() {
+        AbstractEventDatabase anotherDatabase =
+                AbstractEventDatabase.getInMemoryInstance(RuntimeEnvironment.systemContext,
+                        "Another name");
+        assertNotEquals(database, anotherDatabase);
+    }
+
+    /**
+     * Testing whether the database actually starts empty.
+     */
+    @Test
+    public void startsEmpty() {
+        final int expectedStartCount = 0;
+        assertEquals(expectedStartCount, database.access().count());
+    }
+
+    /**
+     * Testing single insertion.
+     */
+    @Test
+    public void canAddOne() {
+        final int expectedCount = 1;
+        database.access().insertEntity(new StringEntity(EXAMPLE));
+        assertEquals(expectedCount, database.access().count());
+    }
+
+    /**
+     * Testing the insertion is actually correct, through {@link AbstractStringDao#contains(String)}.
+     */
+    @Test
+    public void canAddOneAndItsCorrect() {
+        database.access().insertEntity(EXAMPLE_ENTITY);
+        assertTrue(database.access().contains(EXAMPLE));
+    }
+
+    /**
+     * Testing multiple insertions.
+     */
+    @Test
+    public void canAddMultiple() {
+        final int expectedCount = 3;
+        database.access().insertEntities(EXAMPLE_COLLECTION);
+        assertEquals(expectedCount, database.access().count());
+    }
+
+    /**
+     * Testing multiple insertions are actually correct, through {@link AbstractStringDao#getAllEntities()}.
+     */
+    @Test
+    public void canAddMultipleAndTheyAreCorrect() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unit tests should not contain more than 1 assert(s).](https://app.codacy.com/manual/CremaLuca/geoCalendar/pullRequest?prid=4822079)